src/main/cli/commandLine.ts

```diff
@@ -0,0 +1,135 @@
+import { RootState } from "readium-desktop/main/redux/states";
+import { CatalogService } from "readium-desktop/main/services/catalog";
+import { URL } from "url";
+
+export interface ICliParam {
+    readonly argv: yargs.Arguments;
+    quit: boolean;
+}
+
+export interface ICli {
+    name: string;
+    fct: (o: ICliParam) => Promise<boolean>;
+    help: string[];
+}
+
+export const cli: ICli[] = [
+    {
+        name: "_",
+        fct: async ({ argv }) => {
+            let publicationOpenRequested = false;
+            let returnValue = true;
+
+            for (const filePath of argv._) {
+                // import and read publication
+                const catalogService = container.get("catalog-service") as CatalogService;
+                const publication = await catalogService.importFile(filePath);
+                const store = container.get("store") as Store<RootState>;
+                if (publication) {
+                    if (!publicationOpenRequested) {
+                        store.dispatch({
+                            type: readerActions.ActionType.OpenRequest,
+                            payload: {
+                                publication: {
+                                    identifier: publication.identifier,
+                                },
+                            },
+                        });
+                        publicationOpenRequested = true;
+                    }
+                } else {
+                    process.stderr.write(`Publication error for "${filePath}"\n`);
+                    returnValue = false;
+                }
+            }
+            return returnValue;
+        },
+        help: [],
+    },
+    {
+        name: "import",
+        fct: async ({ argv }) => {
+            const catalogService = container.get("catalog-service") as CatalogService;
+            await catalogService.importFile(argv.import as string);
+            return true;
+        },
+        help: [
+            "--import PATH",
+            "import epub or lpcl file",
+        ],
+    },
+    {
+        name: "opds",
+        fct: async ({ argv }) => {
+            // extract and save the title and url from opdsFeed
+            // title=http://myurl.com or get TLD and set url
+            const feed = (argv.opds as string).split("=");
+            const url = feed.length === 2 ? feed[1] : feed[0];
+            const hostname = (new URL(url)).hostname;
+            const title = feed.length === 2 ? feed[0] : hostname;
+            if (hostname) {
+                const opdsRepository = container.get("opds-feed-repository") as OpdsFeedRepository;
+                await opdsRepository.save({ title, url });
+                return true;
+            }
+            return false;
+        },
+        help: [
+            "--opds TITLE=URL | URL",
+            "import opds feed url",
+        ],
+    },
+    {
+        name: "silent",
```

Comment:
Tests (not edge cases, just normal use cases):

1) `npm run cli -- /PATH/TO/EPUB --silent` opens the publication (reader UI) and then forcefully kills the app with `app.exit()`. Note that in this PR the forceful killing occurs immediately after the app opens the publication (too quick to see), so for my test I added a kill-timeout, to actually **see** the UI being opened.
2) `npm run cli -- --silent /PATH/TO/EPUB` kills the app, but never opens the publication => so it looks like the order of parameters matters, but this seems to be a bug, not a feature.
3) `npm run cli -- --silent=true /PATH/TO/EPUB` => same as (1)
4) `npm run cli -- --silent true /PATH/TO/EPUB` => same as (1)
5) `npm run cli -- --silent=zzzz /PATH/TO/EPUB` => same as (1)
6) `npm run cli -- --silent zzzz /PATH/TO/EPUB` => same as (1)
7) `npm run cli -- --silent=false /PATH/TO/EPUB` => same as (1). Obvious bug: `silent=false` should definitely not set `param.quit=true` and subsequently kill the app.
8) the same tests can be run with `--read=TITLE`, for example `npm run cli -- --silent=false --read=Accessibility` (successfully opens "Accessible EPU3", then forcefully kills the app)

The CLI handling code in this PR seems quite fragile. This is why I suggested using a library (like Meow) which handles CLI param types, aliases, ordering, precedence, uncaught exceptions, etc. etc.

Reply:
Why meow is better than yargs ?
https://npmcompare.com/compare/commander,meow,minimist,yargs

yargs is already present

Reply:
But, the choice of library is just an implementation detail. What matters in this PR is the design, currently flawed in terms of the state chart / flow of actions (read-UI + silent -> force-kill, etc.).

There are also bugs like `--silent=false` => `silent` is in fact true in the code! ... this is why I propose Meow which would clarify / simplify the code to ensure correct types and default value handling.

Reply:
Closing this comment, as the code has now changed significantly. There is now proper Args handling of command line parameters, but we will have to test parameter types (e.g. `silent` boolean, `--silent` vs. `--silent=true` vs. `--silent 1`, etc.), default parameter values, parameter combinations and ordering (e.g. `silent` with `read`), etc.